src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -206,7 +206,9 @@ public abstract class Index : ITombstoneAware, IDisposable, ILowMemoryHandler
 
         internal bool IsLowMemory => _lowMemoryFlag.IsRaised();
 
-        protected Index(IndexType type, IndexDefinitionBase definition)
+        private readonly double _txAllocationsRatio;
+
+        protected Index(IndexType type, IndexDefinitionBase definition, bool hasOutputReduceToCollection)
```

Comment:
Instead of adding additional parameter I would prefer to use `definition` to detect if an index has `OutputReduceToCollection` defined. It would just require to cast it to `MapReduceIndexDefinition` when you're inside  `if (type.IsMapReduce())`

Reply:
https://github.com/ravendb/ravendb/pull/9868